packages/server/src/transformer.ts

```diff
@@ -1,11 +1,12 @@
+/* eslint-disable @typescript-eslint/naming-convention */
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
 /**
  * @public
  */
 export type DataTransformer = {
-  serialize(object: any): any;
-  deserialize(object: any): any;
+  serialize<R = any>(object: any): Exclude<R, PromiseLike<any>>;
```

Comment:
Why `PromiseLike` instead of `Promise`? Genuinely curious.

Reply:
I'm not sure what the exact differences are, I know that PromiseLike is a type that extends then,catch, etc (Promise functions basically),
but ofc this could be done using Promise instead of PromiseLike aswell. 

```ts
export type DataTransformer = {
  serialize<R extends any = object>(
    object: any
  ): R extends Promise<infer _U> ? never : R;
  deserialize<R extends any = object>(
    object: any
  ): R extends Promise<infer _U> ? never : R;
};
```

They both essentially are the same and work.

Reply:
I like that better, but I see that it's a minor thing

Reply:
Ye me too, updated in

https://github.com/OrJDev/trpc/commit/016d07d321c6fbbc84942e40c4c6d93b426e1d59